pkg/client/shub/util.go

```diff
@@ -54,19 +54,16 @@ func shubParseReference(src string) (uri ShubURI, err error) {
 	} else if l < 2 {
 		return ShubURI{}, errors.New("Not a valid Shub reference")
 	}
-
-	// look for an @ and split if it exists
-	if strings.Contains(src, `@`) {
-		refParts = strings.Split(src, `@`)
-		uri.digest = `@` + refParts[1]
+	// look for a : and split if it exists
+	if strings.Contains(src, ":") {
+		refParts = strings.Split(src, ":")
+		uri.tag = ":" + refParts[1]
 		src = refParts[0]
 	}
 
-	// look for a : and split if it exists
-	if strings.Contains(src, `:`) {
-		refParts = strings.Split(src, `:`)
-		uri.tag = `:` + refParts[1]
-		src = refParts[0]
+	// look for an @ and split if it exists
```

Comment:
I'm not quite sure if the comment is wrong or the code is wrong.

The code is looking for @ and replacing it with `:commit:`, so foo@bar becomes foo:commit:bar, right?

Reply:
yeah, thats correct. I still need to fix the comments to make sense.

Reply:
@WestleyK will you update this PR?

Reply:
@sashayakovtseva, Yes, soon-ish. But there is still some testing/comparing with 2.X before this can be merged.